src/extension/wps/src/main/java/au/org/emii/wps/gogoduck/index/FeatureSourceIndexReader.java

```diff
@@ -20,23 +22,27 @@
 
 import java.io.IOException;
 import java.net.URI;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
 
 public class FeatureSourceIndexReader implements IndexReader {
     private static final Logger logger = LoggerFactory.getLogger(FeatureSourceIndexReader.class);
+    private final URLMangler urlMangler;
 
     private Catalog catalog = null;
 
-    public FeatureSourceIndexReader(Catalog catalog) {
+    public FeatureSourceIndexReader(Catalog catalog, Map<String, String> urlSubstitution) {
         this.catalog = catalog;
+        this.urlMangler = new URLMangler(urlSubstitution);
     }
 
     @Override
-    public URIList getUriList(String profile, String timeField, String sizeField, String urlField, GoGoDuckSubsetParameters subset) throws GoGoDuckException {
+    public Set<DownloadRequest> getDownloadList(String profile, String timeField, String sizeField, String urlField, SubsetParameters subset) throws GoGoDuckException {
+        Set<DownloadRequest> result = new LinkedHashSet<>();
 
-        URIList uriList = new URIList();
-
-        String timeCoverageStart = subset.get("TIME").start;
-        String timeCoverageEnd = subset.get("TIME").end;
+        String timeCoverageStart = subset.getTimeRange().getStart().toString();
+        String timeCoverageEnd = subset.getTimeRange().getEnd().toString();
 
         // TODO Should include also workspace, but works also without
         String typeName = profile;
```

Comment:
Hah, didn't change this but yes there's no reason to confuse everyone by having two names for something.